rosbridge_library/src/rosbridge_library/protocol.py

```diff
@@ -46,15 +46,26 @@ def is_number(s):
         return True
     except ValueError:
         return False
-        
-def has_binary(d):
-    if type(d)==bson.Binary:
+
+
+def has_binary(obj):
+    """ Returns True if obj is a binary or contains a binary attribute
+    """
+
+    if type(obj) is bson.binary.Binary:
```

Comment:
Throughout: Please use `isinstance()` instead of `type()` (see https://stereochro.me/ideas/type-vs-isinstance)

Reply:
+1